spark/src/main/java/org/opensearch/sql/spark/cluster/FlintStreamingJobHouseKeeperTask.java

```diff
@@ -0,0 +1,161 @@
+  private final FlintIndexMetadataService flintIndexMetadataService;
+  private final StateStore stateStore;
+  private final EMRServerlessClientFactory emrServerlessClientFactory;
+
+  private static final Logger LOGGER = LogManager.getLogger(FlintStreamingJobHouseKeeperTask.class);
+  protected static final AtomicBoolean isRunning = new AtomicBoolean(false);
+
+  @Override
+  public void run() {
+    if (!isRunning.compareAndSet(false, true)) {
+      LOGGER.info("Previous task is still running. Skipping this execution.");
+      return;
+    }
+    try {
+      LOGGER.info("Starting the cleaner task for disabled and deleted data sources.");
+      Map<String, FlintIndexMetadata> autoRefreshFlintIndicesMap = getAllAutoRefreshIndices();
+      autoRefreshFlintIndicesMap.forEach(
+          (autoRefreshIndex, flintIndexMetadata) -> {
+            try {
+              String datasourceName = getDataSourceName(flintIndexMetadata);
+              try {
+                DataSourceMetadata dataSourceMetadata =
+                    this.dataSourceService.getDataSourceMetadata(datasourceName);
+                if (dataSourceMetadata.getStatus() == DataSourceStatus.DISABLED) {
+                  LOGGER.info("Datasource is disabled for autoRefreshIndex: {}", autoRefreshIndex);
+                  alterAutoRefreshIndex(autoRefreshIndex, flintIndexMetadata, datasourceName);
+                } else {
+                  LOGGER.debug("Datasource is enabled for autoRefreshIndex : {}", autoRefreshIndex);
+                }
+              } catch (DataSourceNotFoundException exception) {
+                // Datasource disabled.
+                LOGGER.info("Datasource is deleted for autoRefreshIndex: {}", autoRefreshIndex);
+                if (!isAlreadyInDeletedState(
+                    autoRefreshIndex, flintIndexMetadata, datasourceName)) {
+                  dropAutoRefreshIndex(autoRefreshIndex, flintIndexMetadata, datasourceName);
+                } else {
+                  LOGGER.debug(
+                      "AutoRefresh index: {} is already in deleted state", autoRefreshIndex);
+                }
+              }
+            } catch (Exception exception) {
+              LOGGER.error(
+                  "Failed to alter/cancel index {}: {}",
+                  autoRefreshIndex,
+                  exception.getMessage(),
+                  exception);
+              Metrics.getInstance()
+                  .getNumericalMetric(MetricName.STREAMING_JOB_HOUSEKEEPER_TASK_FAILURE_COUNT)
+                  .increment();
+            }
+          });
+    } catch (Throwable error) {
+      LOGGER.error("Error while running the streaming job cleaner task: {}", error.getMessage());
+      Metrics.getInstance()
+          .getNumericalMetric(MetricName.STREAMING_JOB_HOUSEKEEPER_TASK_FAILURE_COUNT)
+          .increment();
+    } finally {
+      isRunning.set(false);
+    }
+  }
+
+  private boolean isAlreadyInDeletedState(
```

Comment:
@dai-chen  can you look into this method once.

Reply:
I cannot remember that is it possible Flint index doesn't have state doc or miss latestID.. @penghuo

Reply:
can we directly call drop index? does drop index validation logic already handle this?

Reply:
FlintIndexOpDrop already validates initial state. also throws error if state doc not found

Reply:
The check is to avoid multiple drop operations on the same flint index in the next run.
Run1: dropped an index.
Run2: Trying to drop the same index unless we check for deleted state. Also we emit error metric because of this.

Other options:
* Do Alter Index in case of deletion as well.
* Catch IllegalStateException from drop index and ignore.

cc: @seankao-az  @penghuo

Reply:
Going with Catching IllegalStateException and Ignoring.